fe/src/main/java/org/apache/doris/alter/MaterializedViewHandler.java

```diff
@@ -549,7 +548,10 @@ private RollupJobV2 createMaterializedViewJob(String mvName, String baseIndexNam
                     }
                     keyStorageLayoutBytes += baseColumn.getType().getStorageLayoutBytes();
                     Column rollupColumn = new Column(baseColumn);
-                    if ((i + 1) <= FeConstants.shortkey_max_column_count
+                    if(changeStorageFormat) {
+                        rollupColumn.setIsKey(baseColumn.isKey());
+                        rollupColumn.setAggregationType(baseColumn.getAggregationType(), true);
```

Comment:
What is about?

Reply:
when create a segmentv2 rollup on duplicate table, the previous code did not handle this specific sutiation, it will create a all key rollup, all value columns will convert to key column